libs/community/langchain_community/document_loaders/sql_database.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, Callable, Dict, Iterator, List, Optional, Union
+
+import sqlalchemy as sa
+
+from langchain_community.docstore.document import Document
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_community.utilities.sql_database import SQLDatabase
+
+
+class SQLDatabaseLoader(BaseLoader):
+    """
+    Load documents by querying database tables supported by SQLAlchemy.
+
+    For talking to the database, the document loader uses the `SQLDatabase`
+    utility from the LangChain integration toolkit.
+
+    Each document represents one row of the result.
+    """
+
+    def __init__(
+        self,
+        query: Union[str, sa.Select],
+        db: SQLDatabase,
+        parameters: Optional[Dict[str, Any]] = None,
+        page_content_mapper: Optional[Callable[..., str]] = None,
+        metadata_mapper: Optional[Callable[..., Dict[str, Any]]] = None,
+        source_columns: Optional[List[str]] = None,
```

Comment:
(not important) I love using Sequence to cover more valid types (e.g., tuple) + denote the promise that the container won't be mutated